### DISCUSSION TITLE
[@types/verror] incompatible with TypeScript 4.6

### DISCUSSION THREAD

**Discussion by benasher44** [DefinitelyTyped/DefinitelyTyped#59093] [Issues with a @types package]
Running tsc v4.6.2 (works in 4.5.5) gives the following result:

```
$ tsc
node_modules/@types/verror/index.d.ts:31:5 - error TS2416: Property 'cause' in type 'VError' is not assignable to the same property in base type 'Error'.
  Property 'message' is missing in type '() => Error | undefined' but required in type 'Error'.

31     cause(): Error | undefined;
       ~~~~~

  node_modules/typescript/lib/lib.es5.d.ts:1023:5
    1023     message: string;
             ~~~~~~~
    'message' is declared here.

node_modules/@types/verror/index.d.ts:31:5 - error TS2425: Class 'Error' defines instance member property 'cause', but extended class 'VError' defines it as instance member function.

31     cause(): Error | undefined;
       ~~~~~


Found 2 errors in the same file, starting at: node_modules/@types/verror/index.d.ts:31

error Command failed with exit code 2.
```

[a comment by an automated account is not shown]

**Comment by schmod** [comment]
This needs to be fixed, but a workaround is to ensure that `esnext` or `es2022` are not present in your `tsconfig.json`'s `lib` array.

**Reply by kirill-konshin** [reply]
> This needs to be fixed, but a workaround is to ensure that `esnext` or `es2022` are not present in your `tsconfig.json`'s `lib` array.

Thank you!

**Reply by acommodari** [reply]
Hey guys 👋 just ran into this issue myself any ETA on a fix?

**Comment by voxpelli** [comment]
The first part of this will be fixed in TS 4.8 (https://github.com/microsoft/TypeScript/pull/49639) and doesn't only affect `@types/verror` but also polyfills for the new Error Cause standard (which supersedes `VError`) such as mine: https://github.com/voxpelli/pony-cause/issues/35

The second part still needs a fix in `@types/verror` though, I opened this one for that: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61199

**Comment by schmod** [comment]
TBH, I also think VError may also need to be updated itself so it can safely interact with es2022 errors.

**Reply by voxpelli** [reply]
@schmod The helpers in my ponyfill for Error Causes handles `VError`style errors as well, easing the transition: https://github.com/voxpelli/pony-cause#stackwithcauses-getting-full-stack-trace-for-error--all-causes

I don't think VError itself will get updated, it feels redundant with this new spec. I did open an issue about compatibility there last year, but with no responses in six months, so I solved it myself instead: https://github.com/TritonDataCenter/node-verror/issues/81
